### DISCUSSION TITLE
Error: Function Runtimes must have a valid version? Deployment issue

### DISCUSSION THREAD

**Discussion by JadamMyers** [vercel/community#2008] [Help] [answered]
### Summary

This is my first time deploying on Vercel. And when I run 'vercel' in my terminal I do get a URL but I also get these errors "
Error: Function Runtimes must have a valid version, for example `now-php@1.0.0`.
invalid_function_runtime: Function Runtimes must have a valid version, for example `now-php@1.0.0`."

I'm not sure what these mean, I updated to the latest version of vercel and I think my vercel.json file is correct. Would someone be able to help me make sense of this?

### Additional information

// vercel.json

{
    "functions": {
      "api/deezer-proxy.js": {
        "runtime": "nodejs14.x"
      }
    },
    "routes": [
      {
        "src": "/deezer-proxy(.*)",
        "dest": "/api/deezer-proxy.js"
      }
    ]
  }

**Comment by amyegan** [comment, marked as answer]
It looks like you have a problem with your function config. The runtime only needs to be specified if [using an unsupported runtime](https://vercel.com/docs/concepts/projects/project-configuration#using-unsupported-runtimes), so you don't need to specify Node.js here. Instead, you can [set the Node.js Version](https://vercel.com/docs/concepts/functions/serverless-functions/runtimes/node-js#node.js-version) on the Project Settings page.

**Reply by btakita** [reply]
Setting the Node.js Version did not work. Neither did setting `"runtime": "nodejs"` as written in the docs.

Could you post how to use node.js version 20 here?

Thanks

**Reply by amyegan** [reply]
You can use Node v20 in your project by selecting `20.x` in the Node.js Version section on the General page in Project Settings. The default runtime is `nodejs` so you usually don't need to specify the runtime for Node.js Serverless Functions in the `/api` directory.
